src/lib/ui/ClimateSettingForm/ClimateSettingScheduleForm.tsx

```diff
@@ -0,0 +1,75 @@
+import classNames from 'classnames'
+import { useState } from 'react'
+import type { ClimateSetting } from 'seamapi'
+
+import type { UseClimateSettingScheduleData } from 'lib/seam/thermostats/climate-setting-schedules/use-climate-setting-schedule.js'
+import { ContentHeader } from 'lib/ui/layout/ContentHeader.js'
+
+import { ClimateSettingScheduleDeviceSelect } from './ClimateSettingScheduleDeviceSelect.js'
+
+export interface ClimateSettingScheduleFormSubmitData {
+  name: string
+  deviceId: string
+  startDate: string
+  endDate: string
+  timezone: string
+  climateSetting: ClimateSetting
+}
+
+export interface ClimateSettingScheduleFormProps {
+  className?: string
+  onBack?: () => void
+  climateSettingSchedule?: NonNullable<UseClimateSettingScheduleData>
+  isSubmitting: boolean
+  onSubmit: (data: ClimateSettingScheduleFormSubmitData) => void
+}
+
+export function ClimateSettingScheduleForm({
+  className,
+  ...props
+}: ClimateSettingScheduleFormProps): JSX.Element | null {
+  return (
+    <div
+      className={classNames('seam-climate-setting-schedule-form', className)}
+    >
+      <Content {...props} />
+    </div>
+  )
+}
+
+function Content({
+  onBack,
+  climateSettingSchedule,
+}: Omit<ClimateSettingScheduleFormProps, 'className'>): JSX.Element {
+  const [_, setDeviceId] = useState<string>(
+    climateSettingSchedule?.device_id ?? ''
+  )
+
+  const [page, setPage] = useState<
+    | 'device_select'
+    | 'default_setting'
+    | 'name_and_time'
+    | 'timezone_select'
+    | 'climate_setting'
+  >('device_select')
```

Comment:
This is how I've done step-by-step flows in the past, is this acceptable here?

Reply:
Neat. If we use the form hook it might have a recommended (similar?) pattern.

Reply:
They suggest using little-state-machine via "Wizard Form / Funnel" on https://react-hook-form.com/advanced-usage

Reply:
I don't think we really need that here since we don't have "real" routing. What do you think? Their example makes more sense since it's jumping between pages